gdx/src/com/badlogic/gdx/Net.java

```diff
@@ -93,20 +93,92 @@
 
 	/** Provides common HTTP methods to use when creating a {@link HttpRequest}.
 	 * <ul>
-	 * <li>GET</li>
-	 * <li>POST</li>
-	 * <li>PUT</li>
-	 * <li>DELETE</li>
-	 * <li>PATCH</li>
-	 * </ul> */
+	 * <li>
+	 * <b>HEAD</b> Asks for a response identical to that of a GET request but without the response body.
+	 * </li>
+	 * <li>
+	 * <b>GET</b> requests a representation of the specified resource.
+	 * Requests using GET should only retrieve data.
+	 * </li>
+	 * <li>
+	 * <b>POST</b> is used to submit an entity to the specified resource,
+	 * often causing a change in state or side effects on the server.
+	 * </li>
+	 * <li>
+	 * <b>PUT</b> replaces all current representations of the target resource with the request payload.
+	 * </li>
+	 * <li>
+	 * <b>PATCH</b> method is used to apply partial modifications to a resource.
+	 * </li>
+	 * <li>
+	 * <b>DELETE</b> deletes the specified resource.
+	 * </li>
+	 * <li>
+	 * <b>OPTIONS</b> is used to describe the communication options for the target resource.
+	 * </li>
+	 * <li>
+	 * <b>CONNECT</b> establishes a tunnel to the server identified by the target resource.
+	 * </li>
+	 * <li>
+	 * <b>TRACE</b> performs a message loop-back test along the path to the target resource.
+	 * </li>
+	 * </ul>
+	 * @author Monarchis
+	 * */
 	public static interface HttpMethods {
-
+		/**
+		 * The HEAD method asks for a response identical to that of a GET request,
+		 * but without the response body.
+		 **/
+		public static final String HEAD = "HEAD";
+
+		/**
+		 * The GET method requests a representation of the specified resource.
+		 * Requests using GET should only retrieve data.
+		 **/
 		public static final String GET = "GET";
+
+		/**
+		 * The POST method is used to submit an entity to the specified resource,
+		 * often causing a change in state or side effects on the server.
+		 **/
 		public static final String POST = "POST";
+
+		/**
+		 * The PUT method replaces all current representations of the target
+		 * resource with the request payload.
+		 **/
 		public static final String PUT = "PUT";
-		public static final String DELETE = "DELETE";
+
+		/**
+		 * The PATCH method is used to apply partial modifications to a resource.
+		 **/
 		public static final String PATCH = "PATCH";
 
+		/**
+		 * The DELETE method deletes the specified resource.
+		 **/
+		public static final String DELETE = "DELETE";
+
+		/**
+		 * The OPTIONS method is used to describe the communication
+		 * options for the target resource.
+		 **/
+		public static final String OPTIONS = "OPTIONS";
+
+		/**
+		 * The CONNECT method establishes a tunnel to the server
+		 * identified by the target resource.
+		 * Not Supported by libgdx yet.
+		 **/
+		public static final String CONNECT = "CONNECT";
+
+		/**
+		 * The TRACE method performs a message loop-back test along
+		 * the path to the target resource.
+		 * Not Supported by libgdx yet.

```

Comment:
Adding things only to say they aren't supported isn't super nice.

Reply:
@MobiDevelop well, then reject this pr

Reply:
I agree. I think better would be to update Net.java to accept any method string also allowing to specify if output is desired or not. After all, GET or POST could both be used with our without stream output, even though not commonly used for GET as opposed to POST.

Reply:
@noblemaster I'm abled to add some of that methods easy, but connection and trace need more efford.

Reply:
Getting Head to work with GWT, but the others are not supported right now, I'll mark it.